### DISCUSSION TITLE
Font size naming

### DISCUSSION THREAD

**Discussion by Temez1** [chakra-ui/chakra-ui#4698] [Q&A] [answered]
Hi!

Is there some reason for using unintuitive text size naming (md, lg, xl, 2xl, ..)?

Eg. Heading would be much more intuitive to have h1,h2, ... as sizes

Also, it's hard to remember the text size between Text and Header, as they are not the same size when using the same size for both. It would then make more sense to have sizes like h1,h2,.. and for text the sm, md, lg.

Also, the API for Header & Text is not coherent and even documented wrongly (This would be an issue but let's get this solved first). Header uses size prop and Text uses fontSize prop (Even the Text's props are defined with size instead of fontSize).

**Comment by primos63** [comment, marked as answer]
Hey Temez1!

I believe the font size naming is patterned after Tailwind which uses text-xs, sm, base, lg, xl, 2xl, etc.  

I'm of the opinion that h1, h2, etc are perhaps less intuitive. What size is an h1, h2, etc?  Certainly their connotation does not apply to normal text as that wouldn't be intuitive.

`size` and `fontSize` are two different props. Text uses `fontSize` not `size`. If one uses `fontSize` on `Heading` and `Text`, then the same font size is applied to each component. `size` is a convenience prop for the responsive sizing as described in the default theme:
```javascript
// from packages/theme/src/components/heading.ts
const sizes: Record<string, SystemStyleObject> = {
  "4xl": {
    fontSize: ["6xl", null, "7xl"],
    lineHeight: 1,
  },
  "3xl": {
    fontSize: ["5xl", null, "6xl"],
    lineHeight: 1,
  },
  "2xl": {
    fontSize: ["4xl", null, "5xl"],
    lineHeight: [1.2, null, 1],
  },
  xl: {
    fontSize: ["3xl", null, "4xl"],
    lineHeight: [1.33, null, 1.2],
  },
  lg: {
    fontSize: ["2xl", null, "3xl"],
    lineHeight: [1.33, null, 1.2],
  },
  md: { fontSize: "xl", lineHeight: 1.2 },
  sm: { fontSize: "md", lineHeight: 1.2 },
  xs: { fontSize: "sm", lineHeight: 1.2 },
}

const defaultProps = {
  size: "xl",
}
```

You are correct about `Text` props. The `size` prop should be changed to `fontSize`.  There is no default theme for `Text`; therefore, no `size` prop exists. One could create one as the `Text` module makes a call to get its styles `const styles = useStyleConfig("Text", props)`.  I also agree that the docs should clearly state what the `size` prop does on `Heading`.

I can understand the frustration. Take the docs for `Heading` props:

> **Props**
> The Heading composes the Box component, so you can pass all Box props.

So then you click on the `Box` link and find that there are no props listed for `Box`. That should at least say that `Box` supports all of the `Style Props` with a link to the `Style Props` section in the docs.

I think the Text and Heading `size` doc problems along with no props on Box, could be put into a Github issue.

**Reply by Temez1** [reply]
The sizing is opinionated & doesn't really matter :D. Glad to know that what the Header size prop actually does.

And what comes to the documentation I agree there is a lot to improve. Like listing all the props and default values somewhere (the box would do 90% of cases already). After going through the documentation again and again I find new props or new ways to do something from some examples. This is because A) I'm quite a beginner with CSS & avoided learning that when I was using MUI. B) The aliases are sometimes confusing but it just takes time to get used to :) Thank god I'm using TS so it saves my ass with the values most of the time.
